From c4fc3f52862b56d6e046ec0fc211e49c95f636ef Mon Sep 17 00:00:00 2001
From: Jaroslav Skarvada <jskarvad@redhat.com>
Date: Fri, 6 Sep 2013 15:58:45 -0400
Subject: [PATCH 15/40] cmake: Fixed FindGnuradio.cmake file for new 3.7
 locations of headers.

This file should actually be removed since it duplicates the functionality of the cmake-recommended GnuradioConfig.cmake file.
---
 cmake/Modules/FindGnuradio.cmake |   93 ++++++++++++++++++++++----------------
 1 file changed, 54 insertions(+), 39 deletions(-)

diff --git a/cmake/Modules/FindGnuradio.cmake b/cmake/Modules/FindGnuradio.cmake
index a6763c5..6410dfa 100644
--- a/cmake/Modules/FindGnuradio.cmake
+++ b/cmake/Modules/FindGnuradio.cmake
@@ -1,29 +1,45 @@
+# Copyright 2013 Free Software Foundation, Inc.
+#
+# This file is part of GNU Radio
+#
+# GNU Radio is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3, or (at your option)
+# any later version.
+#
+# GNU Radio is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with GNU Radio; see the file COPYING.  If not, write to
+# the Free Software Foundation, Inc., 51 Franklin Street,
+# Boston, MA 02110-1301, USA.
+
 INCLUDE(FindPkgConfig)
 INCLUDE(FindPackageHandleStandardArgs)
 
-# a subset of required modules can be set using the following syntax
-#set(GR_REQUIRED_MODULES RUNTIME BLOCKS)
-
-# if GR_REQUIRED_MODULES is not defined, it will be set to the following list (all of them)
-if(NOT GR_REQUIRED_MODULES)
-    set(GR_REQUIRED_MODULES RUNTIME ANALOG ATSC AUDIO BLOCKS CHANNELS DIGITAL FEC FFT FILTER NOAA PAGER QTGUI TRELLIS UHD VOCODER WAVELET PMT)
+# if GR_REQUIRED_COMPONENTS is not defined, it will be set to the following list (all of them)
+if(NOT GR_REQUIRED_COMPONENTS)
+  set(GR_REQUIRED_COMPONENTS RUNTIME ANALOG BLOCKS DIGITAL FFT FILTER PMT)
 endif()
 
 set(GNURADIO_ALL_LIBRARIES "")
 set(GNURADIO_ALL_INCLUDE_DIRS "")
 
 MACRO(LIST_CONTAINS var value)
-    SET(${var})
-    FOREACH(value2 ${ARGN})
-        IF (${value} STREQUAL ${value2})
-            SET(${var} TRUE)
-        ENDIF(${value} STREQUAL ${value2})
-    ENDFOREACH(value2)
+  SET(${var})
+  FOREACH(value2 ${ARGN})
+    IF (${value} STREQUAL ${value2})
+      SET(${var} TRUE)
+    ENDIF(${value} STREQUAL ${value2})
+  ENDFOREACH(value2)
 ENDMACRO(LIST_CONTAINS)
 
 function(GR_MODULE EXTVAR PCNAME INCFILE LIBFILE)
 
-    LIST_CONTAINS(REQUIRED_MODULE ${EXTVAR} ${GR_REQUIRED_MODULES})
+    LIST_CONTAINS(REQUIRED_MODULE ${EXTVAR} ${GR_REQUIRED_COMPONENTS})
     if(NOT REQUIRED_MODULE)
         #message("Ignoring GNU Radio Module ${EXTVAR}")
         return()
@@ -36,18 +52,18 @@ function(GR_MODULE EXTVAR PCNAME INCFILE LIBFILE)
 
     set(INCVAR_NAME "GNURADIO_${EXTVAR}_INCLUDE_DIRS")
     set(LIBVAR_NAME "GNURADIO_${EXTVAR}_LIBRARIES")
-    set(PC_INCDIR "PC_GNURADIO_${EXTVAR}_INCLUDEDIR")
-    set(PC_LIBDIR "PC_GNURADIO_${EXTVAR}_LIBDIR")
+    set(PC_INCDIR ${PC_GNURADIO_${EXTVAR}_INCLUDEDIR})
+    set(PC_LIBDIR ${PC_GNURADIO_${EXTVAR}_LIBDIR})
 
     # look for include files
     FIND_PATH(
         ${INCVAR_NAME}
         NAMES ${INCFILE}
-        HINTS $ENV{GNURADIO_RUNTIME_DIR}/include/gnuradio
+        HINTS $ENV{GNURADIO_RUNTIME_DIR}/include
             ${PC_INCDIR}
-            ${CMAKE_INSTALL_PREFIX}/include/gnuradio
-        PATHS /usr/local/include/gnuradio
-              /usr/include/gnuradio
+            ${CMAKE_INSTALL_PREFIX}/include
+        PATHS /usr/local/include
+              /usr/include
     )
 
     # look for libs
@@ -85,24 +101,23 @@ function(GR_MODULE EXTVAR PCNAME INCFILE LIBFILE)
 
 endfunction()
 
-GR_MODULE(CORE gnuradio-runtime gr_top_block.h gnuradio-runtime)
-GR_MODULE(RUNTIME gnuradio-runtime gr_top_block.h gnuradio-runtime)
-GR_MODULE(ANALOG gnuradio-analog analog/noise_type.h gnuradio-analog)
-GR_MODULE(ATSC gnuradio-atsc atsc_api.h gnuradio-atsc)
-GR_MODULE(AUDIO gnuradio-audio audio/sink.h gnuradio-audio)
-GR_MODULE(BLOCKS gnuradio-blocks blocks/delay.h gnuradio-blocks)
-GR_MODULE(CHANNELS gnuradio-channels channels/channel_model.h gnuradio-channels)
-GR_MODULE(DIGITAL gnuradio-digital digital/lfsr.h gnuradio-digital)
-GR_MODULE(FEC gnuradio-fec fec/rs.h gnuradio-fec)
-GR_MODULE(FFT gnuradio-fft fft/fft.h gnuradio-fft)
-GR_MODULE(FILTER gnuradio-filter filter/fir_filter.h gnuradio-filter)
-GR_MODULE(NOAA gnuradio-noaa noaa/hrpt.h gnuradio-noaa)
-GR_MODULE(PAGER gnuradio-pager noaa/flex_deinterleave.h gnuradio-pager)
-GR_MODULE(QTGUI gnuradio-qtgui qtgui/utils.h gnuradio-qtgui)
-GR_MODULE(TRELLIS gnuradio-trellis trellis/fsm.h gnuradio-trellis)
-GR_MODULE(UHD gnuradio-uhd gr_uhd/usrp_sink.h gnuradio-uhd)
-GR_MODULE(VOCODER gnuradio-vocoder vocoder/alaw_encode_sb.h gnuradio-vocoder)
-GR_MODULE(WAVELET gnuradio-wavelet wavelet/wavelet_ff.h gnuradio-wavelet)
+GR_MODULE(RUNTIME gnuradio-runtime gnuradio/top_block.h gnuradio-runtime)
+GR_MODULE(ANALOG gnuradio-analog gnuradio/analog/api.h gnuradio-analog)
+GR_MODULE(ATSC gnuradio-atsc gnuradio/atsc/api.h gnuradio-atsc)
+GR_MODULE(AUDIO gnuradio-audio gnuradio/audio/api.h gnuradio-audio)
+GR_MODULE(BLOCKS gnuradio-blocks gnuradio/blocks/api.h gnuradio-blocks)
+GR_MODULE(CHANNELS gnuradio-channels gnuradio/channels/api.h gnuradio-channels)
+GR_MODULE(DIGITAL gnuradio-digital gnuradio/digital/api.h gnuradio-digital)
+GR_MODULE(FCD gnuradio-fcd gnuradio/fcd_api.h gnuradio-fcd)
+GR_MODULE(FEC gnuradio-fec gnuradio/fec/api.h gnuradio-fec)
+GR_MODULE(FFT gnuradio-fft gnuradio/fft/api.h gnuradio-fft)
+GR_MODULE(FILTER gnuradio-filter gnuradio/filter/api.h gnuradio-filter)
+GR_MODULE(NOAA gnuradio-noaa gnuradio/noaa/api.h gnuradio-noaa)
+GR_MODULE(PAGER gnuradio-pager gnuradio/pager/api.h gnuradio-pager)
+GR_MODULE(QTGUI gnuradio-qtgui gnuradio/qtgui/api.h gnuradio-qtgui)
+GR_MODULE(TRELLIS gnuradio-trellis gnuradio/trellis/api.h gnuradio-trellis)
+GR_MODULE(UHD gnuradio-uhd gnuradio/uhd/api.h gnuradio-uhd)
+GR_MODULE(VOCODER gnuradio-vocoder gnuradio/vocoder/api.h gnuradio-vocoder)
+GR_MODULE(WAVELET gnuradio-wavelet gnuradio/wavelet/api.h gnuradio-wavelet)
+GR_MODULE(WXGUI gnuradio-wxgui gnuradio/wxgui/api.h gnuradio-wxgui)
 GR_MODULE(PMT gnuradio-pmt pmt/pmt.h gnuradio-pmt)
-
-
-- 
1.7.10.4

